lib/fluent/plugin/out_google_cloud.rb

```diff
@@ -449,14 +445,14 @@ module InternalConstants
     config_param :monitoring_type, :string,
                  :default => Monitoring::PrometheusMonitoringRegistry.name
 
-    # Whether to call metadata agent to retrieve monitored resource.
+    # Whether to call metadata agent to retrieve monitored resource. This flag
+    # is kept for backwards compatibility, and is no longer used.
+    # TODO: Breaking change. Remove this flag in Logging Agent 2.0.0 release.
     config_param :enable_metadata_agent, :bool, :default => false
```

Comment:
Also, might want to add `:deprecated` and `:skip_accessor` (see https://docs.fluentd.org/plugin-development/api-config-types). Also for `metadata_agent_url`.